### ISSUE TITLE
False positive `no-name-in-module` in Matplotlib component

### ISSUE THREAD

**Issue by JCGoran** [PyCQA/pylint#5655]
### Bug description

Create a file `example.py` with the following content:

```python
from matplotlib.backends.backend_agg import FigureCanvas
```

### Command used

```shell
pylint example.py
```

### Pylint output

```shell
************* Module example
example.py:1:0: C0114: Missing module docstring (missing-module-docstring)
example.py:1:0: E0611: No name 'FigureCanvas' in module 'matplotlib.backends.backend_agg' (no-name-in-module)
example.py:1:0: W0611: Unused FigureCanvas imported from matplotlib.backends.backend_agg (unused-import)

----------------------------------------------------------------------
Your code has been rated at -60.00/10 (previous run: -60.00/10, +0.00)
```

### Expected behavior

Pylint shouldn't report the `No name 'FigureCanvas' in module 'matplotlib.backends.backend_agg' (no-name-in-module)
` error since the class actually exists, and the very next line says that the (now successfully imported) class is unused.

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.2
Python 3.9.9 (main, Dec 16 2021, 23:13:29) 
[GCC 11.2.0]
```

### OS / Environment

Debian Sid (unstable) with GNU/Linux `5.10.0-8-amd64`

### Additional dependencies

```plaintext
matplotlib==3.5.1
matplotlib-inline==0.1.3
pylint==2.12.2
```
